### PR TITLE
Add min, max calculation for normal and tangent in vertex buffer

### PR DESCRIPTION
**Pull request by JoneLau**

### CODE REVIEW

**Comment by JoneLau** [comment]
update cocos-creator/engine-3d#130

File: Assets/Editor/Exporter.Mesh.cs [hunk 1]
```diff
@@ -180,6 +180,10 @@ public partial class Exporter {
       int offsetBuffer = _bufInfo.data.Length;
       Vector3 minPos = Vector3.zero;
       Vector3 maxPos = Vector3.zero;
+      Vector3 minNor = Vector3.zero;
```
Comment (jwu) [inline Assets/Editor/Exporter.Mesh.cs]: minNor => minNormal

File: Assets/Editor/Exporter.Mesh.cs [hunk 2]
```diff
@@ -180,6 +180,10 @@ public partial class Exporter {
...
+      Vector3 maxNor = Vector3.zero;
```
Comment (jwu) [inline Assets/Editor/Exporter.Mesh.cs]: maxNor => maxNormal

File: Assets/Editor/Exporter.Mesh.cs [hunk 3]
```diff
@@ -180,6 +180,10 @@ public partial class Exporter {
...
+      Vector4 minTan = Vector4.zero;
```
Comment (jwu) [inline Assets/Editor/Exporter.Mesh.cs]: minTan => minTangent

File: Assets/Editor/Exporter.Mesh.cs [hunk 4]
```diff
@@ -263,18 +267,64 @@ public partial class Exporter {
...
+              if (norl.y > maxNor.y) {
+                maxNor.y = norl.y;
+              }
+              if (norl.z > maxNor.z) {
+                maxNor.z = norl.z;
+              }
             }
 
             if (tangents.Length > 0) {
+              Vector4 tang = tangents[i];
```
Comment (jwu) [inline Assets/Editor/Exporter.Mesh.cs]: tang => tangent

File: Assets/Editor/Exporter.Mesh.cs [hunk 5]
```diff
@@ -263,18 +267,64 @@ public partial class Exporter {
             }
 
             if (normals.Length > 0) {
+              Vector3 norl = normals[i];
```
Comment (jwu) [inline Assets/Editor/Exporter.Mesh.cs]: norl => normal

File: Assets/Editor/Exporter.Mesh.cs [hunk 6]
```diff
@@ -411,6 +461,8 @@ public partial class Exporter {
           compType = ComponentType.FLOAT32,
           attrType = AttrType.VEC3,
           // TODO: min, max
```
Comment (jwu) [inline Assets/Editor/Exporter.Mesh.cs]: remove this comment

File: Assets/Editor/Exporter.Mesh.cs [hunk 7]
```diff
@@ -425,6 +477,8 @@ public partial class Exporter {
           compType = ComponentType.FLOAT32,
           attrType = AttrType.VEC4,
           // TODO: min, max
```
Comment (jwu) [inline Assets/Editor/Exporter.Mesh.cs]: remove this comment

File: Assets/Editor/Exporter.Mesh.cs [hunk 8]
```diff
@@ -263,18 +267,64 @@ public partial class Exporter {
...
             if (normals.Length > 0) {
+              Vector3 norl = normals[i];
+              norl.z = -norl.z;
               // NOTE: convert LH to RH
-              writer.Write(normals[i].x);
-              writer.Write(normals[i].y);
-              writer.Write(-normals[i].z);
+              writer.Write(norl.x);
+              writer.Write(norl.y);
+              writer.Write(norl.z);
```
Comment (jwu) [inline Assets/Editor/Exporter.Mesh.cs]: leave one line

File: Assets/Editor/Exporter.Mesh.cs [hunk 9]
```diff
@@ -263,18 +267,64 @@ public partial class Exporter {
...
+              tang.z = -tang.z;
               // NOTE: convert LH to RH
-              writer.Write(tangents[i].x);
-              writer.Write(tangents[i].y);
-              writer.Write(-tangents[i].z);
-              writer.Write(tangents[i].w);
+              writer.Write(tang.x);
+              writer.Write(tang.y);
+              writer.Write(tang.z);
+              writer.Write(tang.w);
```
Comment (jwu) [inline Assets/Editor/Exporter.Mesh.cs]: leave one line

File: Assets/Editor/Exporter.Mesh.cs [hunk 10]
```diff
@@ -263,18 +267,68 @@ public partial class Exporter {
...
+              if (normal.y > maxNormal.y) {
+                maxNormal.y = normal.y;
+              }
+              if (normal.z > maxNormal.z) {
+                maxNormal.z = normal.z;
+              }
             }
 
             if (tangents.Length > 0) {
+              Vector4 tang = tangents[i];
```
Comment (jwu) [inline Assets/Editor/Exporter.Mesh.cs]: tang => tangent

**Pull request merged**